src/library-authoring/add-content/AddContent.tsx

```diff
@@ -119,27 +131,51 @@ const AddContentView = ({
   const extraFilter = unitId ? ['NOT block_type = "unit"', 'NOT type = "collections"'] : undefined;
   const visibleTabs = unitId ? [LibraryContentTypes.components] : undefined;
 
+  /** List container content types that should be displayed based on current path */
+  const visibleContentTypes = useMemo(() => {
+    if (insideCollection) {
+      // except for add collection button, show everthing.
+      return [
+        libraryContentButtonData,
+        sectionButtonData,
+        subsectionButtonData,
+        unitButtonData,
+      ];
+    }
+    if (insideUnit) {
+      // Only show libraryContentButton
+      return [libraryContentButtonData];
+    }
+    if (insideSection) {
+      // Should only allow adding subsections
+      return [subsectionButtonData];
```

Comment:
To fix coverage and not let us forget it later:
```suggestion
    //  istanbul ignore if: not implemented
    if (insideSection) {
      // Should only allow adding subsections
      throw new Error('Not implemented');
      // return [subsectionButtonData];
```